third_party/gRPC_utils/proto_reflection_descriptor_database.cc

```diff
@@ -31,6 +31,7 @@ using grpc::reflection::v1alpha::ServerReflection;
 using grpc::reflection::v1alpha::ServerReflectionRequest;
 using grpc::reflection::v1alpha::ServerReflectionResponse;
 
+const uint8_t g_timeoutGrpcMainStreamSeconds = 10; //using default gwhisper timeout of 10 seconds.
```

Comment:
I have a new idea:
gWhisper uses reflection only BEFORE running the actual user-requested RPC:

1. gWhisper starts parsing CLI arguments until hostname
2. once hostname is known, reflection stream is established with the server and via multiple requests all the RPC names and type information is fetched from the reflection endpoint while CLI args are parsed and verified against this.
3. The parsed CLI args are converted into a protobuf message
4. The actual RPC is called as requested with the user (may be streaming or unary RPC)
5. response is formatted and printed

I would need to verify, but I guess that currently, we keep the reflection stream open for the whole duration. which causes problems when using a fixed reflection deadline.

What we could do is close the reflection stream after 2 or 3. This way we could set a fixed reflection stream RPC timeout and we would not influence 4, even if the user executes a very long-running RPC.

Reply:
Hi @rainerschoe thanks for the review.  I did see some deficiencies via functional tests and I like your idea of closing the desc stream early.

Hence, in the new PR
- Dynamic deadlines are now propagated to the DescDb stream.
- The db stream stream is now closed after 2.